WiseUnpacker/SabreTools.Serialization.Deserializers/WiseSectionHeader.cs

```diff
@@ -87,39 +87,193 @@ public class WiseSectionHeader : BaseBinaryDeserializer<SectionHeader>
             //Seek back to the beginning of the section
             data.Seek(initialOffset, 0);
 
-            header.UnknownValue0 = data.ReadUInt32LittleEndian();
+            header.UnknownDataSize = data.ReadUInt32LittleEndian();
             header.SecondExecutableFileEntryLength = data.ReadUInt32LittleEndian();
             header.UnknownValue2 = data.ReadUInt32LittleEndian();
             header.UnknownValue3 = data.ReadUInt32LittleEndian();
             header.UnknownValue4 = data.ReadUInt32LittleEndian();
             header.FirstExecutableFileEntryLength = data.ReadUInt32LittleEndian();
             header.MsiFileEntryLength = data.ReadUInt32LittleEndian();
-            if (headerLength == 6)
-                return header;
+            if (headerLength != 6)
```

Comment:
Recommendation: Have separate blocks for each header length; it'll be significantly easier to read. For example:

```c#
// Read common values

if (headerLength > 6)
{
    // Read values
}

if (headerLength > 8)
{
    // Read values
}

// Read post-header things
```

Reply:
I don't know why I didn't do it this way in the first place, thanks.